### PR TITLE
Ds 368 add forward ref to all yoga components

### PR DESCRIPTION
**Pull request by guilhermecardoso-gympass**
[JIRA Issue](https://gympass.atlassian.net/browse/)

## Description 📄

[Add React.forwardRef on yoga components web/mobile]

## Platforms 📲

- [x] Web
- [x] Mobile

## Type of change 🔍

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] This change is a refactor on exists components 

## How Has This Been Tested? 🧪

[Enter the tips to test this PR]

- [ ] Unit Test
- [ ] Snapshot Test

## Checklist: 🔍

- [x] My code follows the contribution guide of this project [Contributing Guide](https://github.com/Gympass/yoga/blob/master/CONTRIBUTING.md)
- [x] Layout matches design prototype: [FIGMA](https://figma.com/file/YOUR_LINK)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Review by frgiovanna — APPROVED** [review]
How about taking this opportunity to add the `forwardRef` on the Inputs too?

**Comment by guilhermecardoso-gympass** [comment]
> How about taking this opportunity to add the `forwardRef` on the Inputs too?

the inputs already have

File: packages/yoga/src/Dropdown/web/Dropdown.jsx [hunk 1]
```diff
@@ -268,79 +268,74 @@ const getSelectedOption = options =>
...
-            disabled={disabled}
-            selected={selectedItem !== null}
-            isOpen={isOpen}
-            label={label}
-            full={full}
-            {...getInputProps()}
-          />
-          <Button
+const Dropdown = React.forwardRef(
+  ({ error, label, disabled, full, options, onChange, ...rest }, ref) => (
```
[the remark this answers is not in the document]
Comment (diegomarcuz) [reply packages/yoga/src/Dropdown/web/Dropdown.jsx:272]: I set the same [idea](https://github.com/gympass/yoga/commit/492288855ff8b9fcad045ca8fcbad0d2bee8532b) below for it.

**Comment by guilhermecardoso-gympass** [comment]
> > > How about taking this opportunity to add the `forwardRef` on the Inputs too?
> > 
> > 
> > the inputs already have
> 
> I was having some warning regarding the `Input` types, because of the `refs` then I checked and noticed `Input` types are missing the `ref`. I set the [fix](https://github.com/gympass/yoga/commit/492288855ff8b9fcad045ca8fcbad0d2bee8532b) because I hadn't seen your PR, what do you think?
> 
> Screen Shot 2022-06-01 at 18 09 46

Ow! Ok understand I`m go Update this PR

File: packages/yoga/src/Dropdown/web/Dropdown.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (guilhermecardoso-gympass) [reply packages/yoga/src/Dropdown/web/Dropdown.jsx:272]: Done

**Comment by guilhermecardoso-gympass** [comment]
> > > How about taking this opportunity to add the `forwardRef` on the Inputs too?
> > 
> > 
> > the inputs already have
> 
> I was having some warning regarding the `Input` types, because of the `refs` then I checked and noticed `Input` types are missing the `ref`. I set the [fix](https://github.com/gympass/yoga/commit/492288855ff8b9fcad045ca8fcbad0d2bee8532b) because I hadn't seen your PR, what do you think?
> 
> Screen Shot 2022-06-01 at 18 09 46

Done

File: packages/yoga/src/Dropdown/web/Dropdown.jsx [hunk 2]
```diff
@@ -301,6 +300,7 @@ const Dropdown = React.forwardRef(
               isOpen={isOpen}
               label={label}
               full={full}
+              ref={ref}
```
Comment (diegomarcuz) [inline packages/yoga/src/Dropdown/web/Dropdown.jsx:303]: One last comment... add like `const inputRef = ref || useRef(null)` so if no ref is present will be handled to null, what do you think

File: packages/yoga/src/Input/native/Number.jsx [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import React from 'react';
 import Input from './Input';
 
-const InputNumber = props => <Input {...props} keyboardType="numeric" />;
+const InputNumber = React.forwardRef((props, ref) => {
+  return <Input ref={ref} {...props} keyboardType="numeric" />;
```
[the remark this answers is not in the document]
Comment (diegomarcuz) [reply packages/yoga/src/Input/native/Number.jsx:5]: no worries.... could you also add to the other `Input` types? like Email, Tel

**Review by diegomarcuz — APPROVED** [review]

**Comment by guilhermecardoso-gympass** [comment]
https://github.com/all-contributors please add @guilhermecardoso-gympass  for code

**Pull request merged**